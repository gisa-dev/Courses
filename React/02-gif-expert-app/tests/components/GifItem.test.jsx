import {render, screen} from '@testing-library/react';
import GifItem from '../../src/components/GifItem';

describe('Test in <GifItem/>', () => {
	const title = 'Images';
	const url = 'https://picsum.photos/200';

	test('should match the snapshot', () => {
		const {container} = render(<GifItem title={title} url={url} />);

		expect(container).toMatchSnapshot();
	});

	test('should display the image with the url and alt indicated', () => {
		render(<GifItem title={title} url={url} />);

		const {src, alt} = screen.getByRole('img');

		expect(src).toBe(url);
		expect(alt).toBe(title);
	});

	test('should display the title in the component', () => {
		render(<GifItem title={title} url={url} />);
		expect(screen.getByText(title)).toBeTruthy();
	});
});
