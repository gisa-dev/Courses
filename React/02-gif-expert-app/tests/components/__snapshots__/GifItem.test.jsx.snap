// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifItem/> should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Images"
      src="https://picsum.photos/200"
    />
    <p>
      Images
    </p>
  </div>
</div>
`;
